torch/testing/_internal/common_utils.py

```diff
@@ -1880,6 +1880,26 @@ def _process_inputs(self, actual, expected, *, id, allow_subclasses):
         return actual, expected
 
 
+class TypedStoragePair(TensorLikePair):
+    """Pair for :class:`torch.storage.TypedStorage` inputs.
+
+    After their deprecation, they no longer act as a regular sequence, since :meth:`torch.storage.TypedStorage.__len__`
+    and :meth:`torch.storage.TypedStorage.__getitem__` are disabled. Thus, we need to convert them to tensors manually.
+    """
+    def __init__(self, actual, expected, *, rtol_override=0.0, atol_override=0.0, **other_parameters):
+        self._check_inputs_isinstance(actual, expected, cls=torch.storage.TypedStorage)
+        super().__init__(actual, expected, **other_parameters)
+        self.rtol = max(self.rtol, rtol_override)
+        self.atol = max(self.atol, atol_override)
+
+    def _to_tensor(self, typed_storage):
+        return torch.tensor(
+            [typed_storage._getitem(idx) for idx in range(typed_storage._size())],
+            dtype=typed_storage.dtype,
+            device=typed_storage.device,
+        )
```

Comment:
@kurtamohler Is there a better way to create a tensor from a typed storage?

Reply:
Yep, a tensor does wrap a storage. You can create a tensor that wraps `typed_storage` by just doing `torch.tensor(typed_storage)`. The storage will not be copied, so if you want the resulting tensor and the input storage to be independent of each other, you can do `torch.tensor(typed_storage.clone())`

Reply:
That only works for `float32` storages:

```pycon
>>> t = torch.rand((), dtype=torch.float64)
>>> s = t.storage()
>>> s
 0.6773810374210006
[torch.storage.TypedStorage(dtype=torch.float64, device=cpu) of size 1]
>>> torch.tensor(s)
RuntimeError: Expected a Storage of type Float or an UntypedStorage, but got Double
```

And we are hitting this exact scenario in 

https://github.com/pytorch/pytorch/blob/1588ea0dbf16f37ce14cfc8764666985c16ccbf9/test/test_serialization.py#L678

Reply:
Oh that's right, because the `dtype` arg to `torch.tensor` is `torch.float` by default. Similar story for the `device` arg. So you can do `torch.tensor(typed_storage, dtype=typed_storage.dtype, device=typed_storage.device)` in general

Reply:
It seems that worked. After 052405a2c610b36d8244716cf88cfac42f9aaa76 the only error left seems to be one dissected in https://github.com/pytorch/pytorch/pull/89557#issuecomment-1340733413.